cli/src/config.rs

```diff
@@ -419,24 +451,41 @@ impl ConfigEnv {
         Ok(())
     }
 
-    /// Sets the directory where repo-specific config file is stored. The path
-    /// is usually `.jj/repo`.
+    /// Sets the directory where the repo-specific config file is stored. The
+    /// path is usually `$REPO/.jj/repo`.
     pub fn reset_repo_path(&mut self, path: &Path) {
+        self.repo_config = Some(SecureConfig::new_repo(path.to_path_buf()));
         self.repo_path = Some(path.to_owned());
-        self.repo_config_path = Some(ConfigPath::new(path.join("config.toml")));
     }
 
-    /// Returns a path to the repo-specific config file.
-    pub fn repo_config_path(&self) -> Option<&Path> {
-        self.repo_config_path.as_ref().map(|p| p.as_path())
+    /// Returns a path to the existing repo-specific config file.
+    fn maybe_repo_config_path(&self, ui: &Ui) -> Result<Option<PathBuf>, CommandError> {
+        Ok(match (&self.repo_config, self.root_config_dir.as_ref()) {
+            (Some(config), Some(root_config_dir)) => {
+                let mut guard = self.rng.lock().unwrap();
+                handle_secure_config_warnings(
+                    ui,
+                    config.maybe_load_config(&mut guard, &root_config_dir.join("repos"))?,
+                )?
+                .config_file
+            }
+            _ => None,
+        })
     }
 
     /// Returns a path to the existing repo-specific config file.
-    fn existing_repo_config_path(&self) -> Option<&Path> {
-        match self.repo_config_path {
-            Some(ref path) if path.exists() => Some(path.as_path()),
+    pub fn repo_config_path(&self, ui: &Ui) -> Result<Option<PathBuf>, CommandError> {
+        Ok(match (&self.repo_config, self.root_config_dir.as_ref()) {
+            (Some(config), Some(root_config_dir)) => {
+                let mut guard = self.rng.lock().unwrap();
+                handle_secure_config_warnings(
+                    ui,
+                    config.load_config(&mut guard, &root_config_dir.join("repos"))?,
+                )?
+                .config_file
+            }
             _ => None,
-        }
+        })
```

Comment:
nit: maybe we can extract helper function like `repo_config_path_with(..., |config, rng, root| config.load_config(rng, root))`, but I don't feel strongly.